File: Lib/Telerik/Examples/QuickStart/Scheduler/CustomWorkTime/MyTypeConverter.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;

namespace Telerik.Examples.WinControls.Scheduler.CustomWorkTime
{
    class MyTypeConverter : TypeConverter
    {
        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
        {
            if (sourceType == typeof(DateTime))
            {
                return true;
            }

            return base.CanConvertFrom(context, sourceType);
        }

        public override object ConvertFrom(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, object value)
        {
            if (value is DateTime)
            {
                return ((DateTime)value).TimeOfDay;
            }

            return base.ConvertFrom(context, culture, value);
        }

        public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType)
        {
            if (destinationType == typeof(string) || destinationType == typeof(TimeSpan) || destinationType == typeof(DateTime))
            {
                return true;
            }

            return base.CanConvertTo(context, destinationType);
        }

        public override object ConvertTo(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, object value, Type destinationType)
        {
            if (destinationType == typeof(string) && context is Telerik.WinControls.UI.GridDataCellElement)
            {
                Telerik.WinControls.UI.GridDataCellElement cell = (Telerik.WinControls.UI.GridDataCellElement)context;
                TimeSpan span = (TimeSpan)value;

                return span.Hours.ToString("00") + ":" + span.Minutes.ToString("00");
            }

            if (destinationType == typeof(TimeSpan))
            {
                return value;
            }

            if (destinationType == typeof(DateTime))
            {
                DateTime dt;
                if (DateTime.TryParse(value.ToString(), out dt))
                {
                    return dt;
                }
            }

            return base.ConvertTo(context, culture, value, destinationType);
        }
    }
}

File: Lib/Telerik/Examples/QuickStart/Scheduler/CustomWorkTime/MyTypeConverter.vb
Imports System.ComponentModel
Imports System.Linq
Imports System.Text

Namespace Telerik.Examples.WinControls.Scheduler.CustomWorkTime
	Friend Class MyTypeConverter
		Inherits TypeConverter
		Public Overrides Overloads Function CanConvertFrom(ByVal context As ITypeDescriptorContext, ByVal sourceType As Type) As Boolean
			If sourceType Is GetType(Date) Then
				Return True
			End If

			Return MyBase.CanConvertFrom(context, sourceType)
		End Function

		Public Overrides Overloads Function ConvertFrom(ByVal context As ITypeDescriptorContext, ByVal culture As System.Globalization.CultureInfo, ByVal value As Object) As Object
			If TypeOf value Is Date Then
				Return (CDate(value)).TimeOfDay
			End If

			Return MyBase.ConvertFrom(context, culture, value)
		End Function

		Public Overrides Overloads Function CanConvertTo(ByVal context As ITypeDescriptorContext, ByVal destinationType As Type) As Boolean
			If destinationType Is GetType(String) OrElse destinationType Is GetType(TimeSpan) OrElse destinationType Is GetType(Date) Then
				Return True
			End If

			Return MyBase.CanConvertTo(context, destinationType)
		End Function

		Public Overrides Overloads Function ConvertTo(ByVal context As ITypeDescriptorContext, ByVal culture As System.Globalization.CultureInfo, ByVal value As Object, ByVal destinationType As Type) As Object
			If destinationType Is GetType(String) AndAlso TypeOf context Is Telerik.WinControls.UI.GridDataCellElement Then
				Dim cell As Telerik.WinControls.UI.GridDataCellElement = CType(context, Telerik.WinControls.UI.GridDataCellElement)
				Dim span As TimeSpan = CType(value, TimeSpan)

				Return span.Hours.ToString("00") & ":" & span.Minutes.ToString("00")
			End If

			If destinationType Is GetType(TimeSpan) Then
				Return value
			End If

			If destinationType Is GetType(Date) Then
				Dim dt As Date
				If Date.TryParse(value.ToString(), dt) Then
					Return dt
				End If
			End If

			Return MyBase.ConvertTo(context, culture, value, destinationType)
		End Function
	End Class
End Namespace